### PR TITLE
feat: new joinSpectraData() function

### PR DESCRIPTION
**Pull request by lgatto**

### CODE REVIEW

**Comment by lgatto** [comment]
This PR addressed #149

File: R/Spectra.R [hunk 1]
```diff
@@ -365,6 +365,18 @@ NULL
 #' - `split`: splits the `Spectra` object based on parameter `f` into a `list`
 #'   of `Spectra` objects.
 #'
+#' - `joinSpectraData`: Individual spectra data variable can be
```
Comment (jorainer) [inline R/Spectra.R:368]: I would suggest to change that sentence to: `individual spectra variables can be...`.

File: R/Spectra.R [hunk 2]
```diff
@@ -365,6 +365,18 @@ NULL
...
+#'    directly added with the `$<-` or `[[<-` syntax. The
+#'    `joinSpectraData()` function allows to merge a `DataFrame` to
+#'    the existing spectra data. This function diverges from the
+#'    `merge()` method in two main ways:
```
Comment (jorainer) [inline R/Spectra.R:372]: `[merge()]` instead of `merge()` to add the link to the `merge` help page.

File: R/Spectra.R [hunk 3]
```diff
@@ -688,11 +700,22 @@ NULL
 #'
 #' @param x A `Spectra` object.
 #'
-#' @param y A `Spectra` object.
+#' @param y A `Spectra` object. A `DataFrame` for `joinSpectraData()`.
+#'
+#' @param by.x A `character(1)` specifying of the spectra variable
```
Comment (jorainer) [inline R/Spectra.R:705]: Please remove the `of`.

File: R/Spectra.R [hunk 4]
```diff
@@ -688,11 +700,22 @@ NULL
...
+#'     used for merging. Default is `"spectrumId"`.
+#'
+#' @param by.y A `character(1)` specifying of the column used for
```
Comment (jorainer) [inline R/Spectra.R:708]: Also here, please remove `of`

**Review by jorainer — CHANGES_REQUESTED** [review]
Great addition, thanks @lgatto . I have some minor change requests in the documentation. Also, could you please add an example of the function to the examples of `Spectra`? And lastly, please bump the version as I have pushed the version already to Bioconductor.

**Comment by lgatto** [comment]
- I had an example ready but forgot to add it in the end. Will be done in next PR.
- Will bump version - hadn't realised it was pushed to Bioc. I would suggest whoever pushed to Bioc bump on Github and mentions 'nothing yet' in the NEWS file.

**Comment by lgatto** [comment]
I also fixed a missing export statement. 

@jorainer - feel free to merge if you are happy with the PR.

**Review by jorainer — APPROVED** [review]
All fine, thanks @lgatto!

**Comment by jorainer** [comment]
And pushed to BioC

**Comment by lgatto** [comment]
There is an issue with checks, which I didn't have locally. I'm investigating.

**Comment by jorainer** [comment]
No worries - it's because the new docker image with Bioconductor 3.13 is not yet working properly (I think the `DelayedArray` package is not available yet).

**Comment by jorainer** [comment]
Actually, seems it's more the `msdata` and the `vdiffr` packages. Install/download of `msdata` fails due to a corrupt download also for me:
```r
> BiocManager::install("msdata")
Bioconductor version 3.13 (BiocManager 1.30.10), R Under development (unstable)
  (2020-10-24 r79367)
Installing package(s) 'msdata'
trying URL 'https://bioconductor.org/packages/3.13/data/experiment/src/contrib/msdata_0.31.0.tar.gz'
Content type 'application/x-gzip' length 446179202 bytes (425.5 MB)
===========================
downloaded 237.2 MB

Error in download.file(url, destfile, method, mode = "wb", ...) : 
  download from 'https://bioconductor.org/packages/3.13/data/experiment/src/contrib/msdata_0.31.0.tar.gz' failed
In addition: Warning messages:
1: In download.file(url, destfile, method, mode = "wb", ...) :
  downloaded length 248742946 != reported length 446179202
2: In download.file(url, destfile, method, mode = "wb", ...) :
  URL 'https://bioconductor.org/packages/3.13/data/experiment/src/contrib/msdata_0.31.0.tar.gz': Timeout of 60 seconds was reached
Warning in download.packages(pkgs, destdir = tmpd, available = available,  :
  download of package ‘msdata’ failed

```

**Comment by lgatto** [comment]
This seems a temporary glitch - I was able to re-install that same version of `msdata` locally:

```r
> packageVersion("msdata")
[1] ‘0.31.0’
> BiocManager::install("msdata")
Bioconductor version 3.13 (BiocManager 1.30.10), R Under development (unstable)
  (2020-04-28 r78328)
Installing package(s) 'msdata'
trying URL 'https://bioconductor.org/packages/3.13/data/experiment/src/contrib/msdata_0.31.0.tar.gz'
Content type 'application/x-gzip' length 446179202 bytes (425.5 MB)
==================================================
downloaded 425.5 MB

* installing *source* package ‘msdata’ ...
** using staged installation
** R
** data
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (msdata)

The downloaded source packages are in
	‘/tmp/RtmpzHuMM8/downloaded_packages’
Installation path not writeable, unable to update packages: codetools,
  KernSmooth
> packageVersion("msdata")
[1] ‘0.31.0’
```

**Comment by jorainer** [comment]
I get the same error again and again (although the size of the downloaded file differs). I'm wondering if this might have to do with the current developmental version of R (i.e. 4.1).

**Pull request merged**